### ISSUE TITLE
Errors on APK Filenames with Spaces in them

### ISSUE THREAD

**Issue by zenware** [AndroBugs/AndroBugs_Framework#7]
I wanted to play with this tool so I just extracted all the packages from my device to run them through and started getting output like this:
```
androbugs.py: error: unrecognized arguments: FileShare Service_com.samsung.android.allshare.service.fileshare.apk
Analyzing APK(10/190): Amazon app suite_com.amazon.fv.apk
usage: androbugs.py [-h] -f APK_FILE [-m ANALYZE_MODE]
                    [-b ANALYZE_ENGINE_BUILD] [-t ANALYZE_TAG] [-e EXTRA]
                    [-c LINE_MAX_OUTPUT_CHARACTERS] [-s] [-v]
                    [-o REPORT_OUTPUT_DIR]
androbugs.py: error: unrecognized arguments: app suite_com.amazon.fv.apk
Analyzing APK(11/190): Amazon Kindle_com.amazon.kindle.apk
usage: androbugs.py [-h] -f APK_FILE [-m ANALYZE_MODE]
                    [-b ANALYZE_ENGINE_BUILD] [-t ANALYZE_TAG] [-e EXTRA]
                    [-c LINE_MAX_OUTPUT_CHARACTERS] [-s] [-v]
                    [-o REPORT_OUTPUT_DIR]
androbugs.py: error: unrecognized arguments: Kindle_com.amazon.kindle.apk
Analyzing APK(12/190): Amazon Music_com.amazon.mp3.apk
usage: androbugs.py [-h] -f APK_FILE [-m ANALYZE_MODE]
                    [-b ANALYZE_ENGINE_BUILD] [-t ANALYZE_TAG] [-e EXTRA]
                    [-c LINE_MAX_OUTPUT_CHARACTERS] [-s] [-v]
                    [-o REPORT_OUTPUT_DIR]
androbugs.py: error: unrecognized arguments: Music_com.amazon.mp3.apk
Analyzing APK(13/190): Amazon Shopping_com.amazon.mShop.android.shopping.apk
usage: androbugs.py [-h] -f APK_FILE [-m ANALYZE_MODE]
                    [-b ANALYZE_ENGINE_BUILD] [-t ANALYZE_TAG] [-e EXTRA]
                    [-c LINE_MAX_OUTPUT_CHARACTERS] [-s] [-v]
                    [-o REPORT_OUTPUT_DIR]
```

Obviously, I can normalize that by replacing the spaces with underscores but I think it would be better if the tool could handle filenames with spaces. So when the massive analysis tool is feeding the list of files to the regular tool I think just adding a backslash to escape whenever the space char is found will be OS agnostic.

Does anyone else have an opinion?

**Comment by AndroBugs** [comment]
It's not an issue.
Use: python androbugs.py -f "The App with Space.apk"

**Comment by zenware** [comment]
I really didn't want to reopen this issue, but... the massive analysis tool is what I was having trouble with. If you just feed it a directory like the instructions say, it throws the error output like I posted before. I think it should automatically handle the spaces. Of course I could write my own script which uses `androbugs.py -f "App With Space.apk"` against the output of `ls ./AllMyApps` which might contain entries like
```
S Finder_com.samsung.android.app.galaxyfinder.apk
SASystemProviders_com.samsung.accessory.safiletransfer.apk
```
Because I'm certainly not going to write them all manually.

But why doesn't the massive analysis tool already do it right?
It comes down to [this line in the code](https://github.com/AndroBugs/AndroBugs_Framework/blob/master/AndroBugs_MassiveAnalysis.py#L103) which reads:
```
cmd = main_cmd + " -s -v -e " + str(args.extra) + " -f " + os.path.join(input_dir, filename) + " -o " + output_dir + " -m " + ANALYZE_MODE_MASSIVE + " -b " + str(args.analyze_engine_build) + " -t " + str(args.analyze_tag)
```

I think instead it should read like this:
```
cmd = main_cmd + " -s -v -e " + str(args.extra) + " -f \"" + os.path.join(input_dir, filename) + "\" -o " + output_dir + " -m " + ANALYZE_MODE_MASSIVE + " -b " + str(args.analyze_engine_build) + " -t " + str(args.analyze_tag)
```

So that's how I have it running in my environment.

**Comment by CalebFenton** [comment]
Cool. Looks like a useful change. PR?

**Comment by zenware** [comment]
Is this project still active whatsoever because I'll try to start making useful changes, otherwise could someone point me in the direction of the most similar active project?

**Comment by AndroBugs** [comment]
@zenware 
This issue will be fixed in the AndroBugs Framework v1.1.0 which will be released within 6 months.
Thanks for reporting!
